docs/tracingpolicy/README.md

```diff
@@ -0,0 +1,1193 @@
+within system calls like `open` or `openat`. It is a good place for tracking
+file descriptor and filename matching.
+
+Let's take a look at the following example:
+```yaml
+- call: "fd_install"
+  syscall: false
+  args:
+  - index: 0
+    type: int
+  - index: 1
+    type: "file"
+  selectors:
+  - matchPIDs:
+      # [...]
+    matchArgs:
+      # [...]
+    matchActions:
+    - action: FollowFD
+      argFd: 0
+      argName: 1
+```
+
+This action uses the dedicated `argFd` and `argName` fields to get respectively
+the index of the file descriptor argument and the index of the name argument in
+the call.
+
+##### UnfollowFD action
+
+The `UnfollowFD` action takes a file descriptor from a system call and deletes
+the corresponding entry from the BPF map, where it was put under the `FollowFD`
+action.
+
+Let's take a look at the following example:
+```yaml
+- call: "__x64_sys_close"
+  syscall: true
+  args:
+  - index: 0
+    type: "int"
+  selectors:
+  - matchPIDs:
+    - operator: NotIn
+      followForks: true
+      isNamespacePID: true
+      values:
+      - 0
+      - 1
+    matchActions:
+    - action: UnfollowFD
+      argFd: 0
+```
+
+Similar to the `FollowFD` action, the index of the file descriptor is described
+under `argFd`:
+```yaml
+matchActions:
+- action: UnfollowFD
+  argFd: 0
+```
+
+In this example, `argFD` is 0. So, the argument from the `__x64_sys_close`
+system call at `index: 0` will be deleted from the BPF map whenever a
+`__x64_sys_close` is executed.
+```yaml
+- index: 0
+  type: "int"
+```
+
+Note, that whenever we would like to follow a file descriptor with a `FollowFD`
+block, there should be a matching `UnfollowFD` block, otherwise the BPF map
+will be broken.
+
+##### CopyFD action
+
+The `CopyFD` action is specific to duplication of file descriptor use cases.
+Similary to `FollowFD`, it takes an `argFd` and `argName` arguments. It can
+typically be used tracking the `dup`, `dup2` or `dup3` syscalls.
+
+See the following example for illustration:
+
+```yaml
+- call: "__x64_sys_dup2"
+  syscall: true
+  args:
+  - index: 0
+    type: "fd"
+  - index: 1
+    type: "int"
+  selectors:
+  - matchPIDs:
+    # [...]
+    matchActions:
+    - action: CopyFD
+      argFd: 0
+      argName: 1
+- call: "__x64_sys_dup3"
+  syscall: true
+  args:
+  - index: 0
+    type: "fd"
+  - index: 1
+    type: "int"
+  - index: 2
+    type: "int"
+  selectors:
+  - matchPIDs:
+    # [...]
+    matchActions:
+    - action: CopyFD
+      argFd: 0
+      argName: 1
+```
+
+##### GetUrl action
+
+The `GetUrl` action can be used to perform a remote interaction such as
+triggering Thinkst canaries or any system that can be triggered via an URL
+request. It uses the `argUrl` field to specify the URL to request using GET
+method.
+
+```yaml
+matchActions:
+- action: GetUrl
+  argUrl: http://ebpf.io
+```
+
+##### DnsLookup action
+
+The `DnsLookup` action can be used to perform a remote interaction such as
```

Comment:
What was the motivation behind DnsLookup and GetUrl? What are the use cases? I think for GetUrl we wanted to send notifications whenever a certain policy gives an event, but I don't know about DnsLookup. Also do we return with the IP address after the lookup or no?

Reply:
The idea is explained here with the Thinkst canaries example, triggering some remote alerting systems by DNS or GET queries https://github.com/cilium/tetragon/pull/433. I don't think we return with the IP address, the point is mostly triggering some stuff and not really about the request itself.